xarray/backends/zarr.py

```diff
@@ -0,0 +1,554 @@
+    return zarr_group
+
+
+def _dask_chunks_to_zarr_chunks(chunks):
+    '''this function dask chunks syntax to zarr chunks'''
+    if chunks is None:
+        return chunks
+
+    all_chunks = product(*chunks)
+    first_chunk = next(all_chunks)
+    for this_chunk in all_chunks:
+        if not (this_chunk == first_chunk):
+            raise ValueError("zarr requires uniform chunk sizes, found %r" %
+                             chunks)
+    return first_chunk
+
+
+def _determine_zarr_chunks(enc_chunks, var_chunks, ndim):
+    """
+    Given encoding chunks (possibly None) and variable chunks (possibly None)
+    """
+
+    # zarr chunk spec:
+    # chunks : int or tuple of ints, optional
+    #   Chunk shape. If not provided, will be guessed from shape and dtype.
+
+    # if there are no chunks in encoding and the variable data is a numpy
+    # array, then we let zarr use its own heuristics to pick the chunks
+    if var_chunks is None and enc_chunks is None:
+        return None
+
+    # if there are no chunks in encoding but there are dask chunks, we try to
+    # use the same chunks in zarr
+    # However, zarr chunks needs to be uniform for each array
+    # http://zarr.readthedocs.io/en/latest/spec/v1.html#chunks
+    # while dask chunks can be variable sized
+    # http://dask.pydata.org/en/latest/array-design.html#chunks
+    if var_chunks and enc_chunks is None:
+        all_var_chunks = product(*var_chunks)
+        first_var_chunk = next(all_var_chunks)
+        for this_chunk in all_var_chunks:
+            if not (this_chunk == first_var_chunk):
+                raise ValueError(
+                    "zarr requires uniform chunk sizes, but variable has "
+                    "non-uniform chunks %r. Consider rechunking the data "
+                    "using `chunk()`." % var_chunks)
+        return first_var_chunk
+
+    # from here on, we are dealing with user-specified chunks in encoding
+    # zarr allows chunks to be an integer, in which case it uses the same chunk
+    # size on each dimension.
+    # Here we re-implement this expansion ourselves. That makes the logic of
+    # checking chunk compatibility easier
+
+    # this coerces a single int to a tuple but leaves a tuple as is
+    enc_chunks_tuple = tuple(enc_chunks)
+    if len(enc_chunks_tuple) == 1:
+        enc_chunks_tuple = ndim * enc_chunks_tuple
+
+    if not len(enc_chunks_tuple) == ndim:
+        raise ValueError("zarr chunks tuple %r must have same length as "
+                         "variable.ndim %g" %
+                         (enc_chunks_tuple, _DIMENSION_KEY))
+
+    for x in enc_chunks_tuple:
+        if not isinstance(x, int):
+            raise ValueError("zarr chunks must be an int or a tuple of ints. "
+                             "Instead found %r" % (enc_chunks_tuple,))
+
+    # if there are chunks in encoding and the variabile data is a numpy array,
+    # we use the specified chunks
+    if enc_chunks_tuple and var_chunks is None:
```

Comment:
I'm not quite sure why you're testing enc_chunks_tuple here. You already guarantee this is a tuple so this would be checking if it's length 0?